test/LaunchDarkly.EventSource.Tests/Stubs/StubHandler.cs

```diff
@@ -1,23 +1,25 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
 using System.Net;
 using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace LaunchDarkly.EventSource.Tests
 {
     public class StubMessageHandler : HttpMessageHandler
```

Comment:
This is what's behind most of the test changes. The way this worked previously was that you would tell it how to respond to each HTTP request by providing a complete HTTP response. There were a couple problems with that: 1. each connection would be closed as soon as the response was sent, which is not an accurate simulation of how event streams normally behave, and also introduced reconnection behavior into every test whether we were testing that or not; 2. it provided no way to introduce a delay into a stream - the best you could do was use the "StubEventSource" class (now deleted) which hacked its own stream reading logic to pretend that there was a delay.

So, now, for successful responses that are supposed to provide a stream, it really provides streaming content and you can specify any number of writes and/or delays that should take place.